messaging.rst

```diff
@@ -159,202 +159,107 @@ Locked Transfer message
-
-After creating the packed form of the data we can use ``keccak256`` to create the ``additional_hash``. ::
-
-    additional_hash = keccak256(packed_message_data)
-
-    0xaeba3609fa01beca6b0b54f49780f301d59a8994bc0b383b2571eee0e6dacad4
-
-3. Balance Hash
+The data will be packed as follows to compute the :term:`additional hash`:
+
++--------------------------------------+---------+-------------+
+| Field                                | Type    | Size (bytes)|
++======================================+=========+=============+
+| command_id (7 for ``LockedTransfer``)| uint8   |   1         |
++--------------------------------------+---------+-------------+
+| message_identifier                   | uint64  |   8         |
++--------------------------------------+---------+-------------+
+| payment_identifier                   | uint64  |   8         |
++--------------------------------------+---------+-------------+
+| expiration                           | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| token_network_address                | address |  20         |
++--------------------------------------+---------+-------------+
+| token                                | address |  20         |
++--------------------------------------+---------+-------------+
+| recipient                            | address |  20         |
++--------------------------------------+---------+-------------+
+| target                               | address |  20         |
++--------------------------------------+---------+-------------+
+| initiator                            | address |  20         |
++--------------------------------------+---------+-------------+
+| secrethash                           | bytes32 |  32         |
++--------------------------------------+---------+-------------+
+| amount                               | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| fee                                  | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| metadata_hash                        | bytes32 |  32         |
++--------------------------------------+---------+-------------+
+
+The computation of the ``metadata_hash`` as well as the exact format of the ``metadata`` itself
+are implementation specific.
```

Comment:
Good question at least for multihop transfers, since the metadata contains the routing information and mediator nodes will not route themselves. So the metadata format is part of the protocol and has to be documented in all it's awkwardness?

Reply:
I've not been involved in any discussions about this format, but yes, I think so.